java/vector/src/main/java/org/apache/arrow/vector/ipc/message/MessageSerializer.java

```diff
@@ -723,8 +723,13 @@ public static MessageMetadataResult readMessage(ReadChannel in) throws IOExcepti
   public static ArrowBuf readMessageBody(ReadChannel in, long bodyLength,
       BufferAllocator allocator) throws IOException {
     ArrowBuf bodyBuffer = allocator.buffer(bodyLength);
```

Comment:
Nice catch!
A more conventional way is to wrap the statements in try-with-resource block?